.github/workflows/checks.yml

```diff
@@ -82,3 +86,29 @@ jobs:
             --platform linux/amd64           \
             ${{ env.DOCKER_IMAGE }}         \
             cargo contract build --manifest-path legacy-shielder/contract/Cargo.toml
+
+  shielder-checks:
+    runs-on: [self-hosted, Linux, X64, large]
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: 1.76.0
+
+      - name: Install WASM target
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Install cargo-contract
+        run: cargo install cargo-contract --version 4.0.0-rc.2
+
+      - name: Run checks for `shielder/contract`
+        run: cd shielder/ && make check-shielder
+
+      - name: Build shielder contract
+        run: cd shielder/ && make build-shielder
```

Comment:
this doesn't make sense, right? since the contract will be built anyway for tests

Reply:
removed, but added build-shielder as dependency for tests

Reply:
still unnecessary, right? drink builds the contract anyway